multiversx_sdk_cli/delegation/staking_provider.py

```diff
@@ -408,14 +394,16 @@ def prepare_transaction_for_setting_metadata(self, owner: IAccount, args: Any) -
 
         return tx
 
-    def prepare_transaction_for_creating_delegation_contract_from_validator(self, owner: IAccount, args: Any) -> ITransaction:
-        receiver = Address.new_from_hex("000000000000000000010000000000000000000000000000000000000004ffff", get_address_hrp()).to_bech32()
+    def prepare_transaction_for_creating_delegation_contract_from_validator(self, owner: IAccount, args: Any) -> Transaction:
+        receiver = Address.new_from_hex("000000000000000000010000000000000000000000000000000000000004ffff", get_address_hrp())
         max_cap = int(args.max_cap)
         fee = int(args.fee)
-        data = "makeNewContractFromValidatorData@" + args_to_string([max_cap, fee])
+
+        serializer = Serializer()
+        data = "makeNewContractFromValidatorData@" + serializer.serialize([BigUIntValue(max_cap), BigUIntValue(fee)])
```

Comment:
Maybe we should have this function in the sdk-py (factories, controllers), as well?

Reply:
indeed, will be added.